language-support/ts/daml-ledger/index.test.ts

```diff
@@ -162,6 +163,26 @@ describe("streamQuery", () => {
     expect(mockChange).toHaveBeenLastCalledWith([fooCreateEvent(1)])
   });
 
+  test("reconnect on close", () => {
+    const ledger = new Ledger({...mockOptions, reconnectThreshold: 0} );
+    const stream = ledger.streamQuery(Foo);
+    stream.on("live", mockLive);
+    stream.on("close", mockClose);
+    mockInstance.serverSend({events: [], offset: '3'});
+    mockConstructor.mockClear();
+    mockInstance.serverClose({code: 1, reason: 'test close'});
+    expect(mockConstructor).toHaveBeenCalled();
+    mockInstance.serverOpen();
+    expect(mockSend).toHaveBeenNthCalledWith(1, {"offset": "3"});
+    expect(mockSend).toHaveBeenNthCalledWith(2, [{"templateIds": ["foo-id"]}]);
+    mockSend.mockClear();
+    mockConstructor.mockClear();
+
+    // check that the client doesn't try to reconnect again.
+    mockInstance.serverClose({code: 1, reason: 'test close'});
```

Comment:
Can you please leave a hint why we are not reconnecting?

Reply:
added a hint.